bellatrix.plugins.jira.zephyr/src/main/java/plugins/jira/zephyr/ZephyrPlugin.java

```diff
@@ -0,0 +1,150 @@
+import plugins.jira.zephyr.events.ZephyrCyclePluginEventArgs;
+import plugins.jira.zephyr.events.ZephyrExecutionPluginEventArgs;
+import plugins.jira.zephyr.services.ZephyrApiService;
+import plugins.jira.zephyr.utilities.DateTimeUtilities;
+import solutions.bellatrix.core.configuration.ConfigurationService;
+import solutions.bellatrix.core.plugins.EventListener;
+import solutions.bellatrix.core.plugins.Plugin;
+import solutions.bellatrix.core.plugins.TestResult;
+import solutions.bellatrix.core.plugins.TimeRecord;
+
+import java.lang.reflect.Method;
+
+import static io.restassured.RestAssured.given;
+
+public class ZephyrPlugin extends Plugin {
+    public static final EventListener<ZephyrCyclePluginEventArgs> ZEPHYR_CYCLE_CREATED = new EventListener<>();
+    public static final EventListener<ZephyrExecutionPluginEventArgs> ZEPHYR_TEST_CASE_EXECUTION_FAILED = new EventListener<>();
+    public static final EventListener<ZephyrExecutionPluginEventArgs> ZEPHYR_TEST_CASE_EXECUTED = new EventListener<>();
+    public static final EventListener<ZephyrCyclePluginEventArgs> ZEPHYR_CYCLE_STATUS_UPDATE_FAILED = new EventListener<>();
+
+    private ZephyrLocalData data;
+
+    public ZephyrPlugin() {
+        String authToken = settings().getToken();
+
+        RestAssured.baseURI = settings().getApiUrl();
+        RestAssured.requestSpecification = new RequestSpecBuilder()
+                .addHeader("Authorization", "Bearer " + authToken)
+                .addHeader("Content-Type", "application/json")
+                .build();
+
+        data = new ZephyrLocalData();
+    }
+
+    private ZephyrSettings settings() {
+        return ConfigurationService.get(ZephyrSettings.class);
+    }
+
+    private boolean isEnabled() {
+        return settings().isEnabled();
+    }
+
+    private String getExecutionId(Method memberInfo) {
+        if (memberInfo.isAnnotationPresent(ZephyrTestCase.class)) {
+            return memberInfo.getAnnotation(ZephyrTestCase.class).id();
+        } else {
+            return "";
+        }
+    }
+
+    private String getProjectId(Method memberInfo) {
+        if (memberInfo.getDeclaringClass().isAnnotationPresent(ZephyrProject.class)) {
+            return memberInfo.getDeclaringClass().getAnnotation(ZephyrProject.class).id();
+        } else {
+            try {
+                return settings().getDefaultProjectKey();
+            } catch (Exception ex) {
+                return "";
+            }
+        }
+    }
+
+    private String getStatus(TestResult testResult) {
+        if (testResult.equals(TestResult.SUCCESS)) {
+            return "Pass";
+        } else {
+            return "Fail";
+        }
+    }
+
+    @Override
+    public void postBeforeClass(Class type) {
+        if (!isEnabled()) return;
+
+        var testCycleName = String.format("%s %s",
+                DateTimeUtilities.getUtcNow(),
+                settings().getTestCycleName().isBlank() ? settings().getTestCycleName() : "BELLATRIX RUN");
```

Comment:
this is fine for now, but potentially we would want to set this via Environment variable.